sklearn/naive_bayes.py

```diff
@@ -117,6 +118,303 @@ def predict_proba(self, X):
+    >>> y = np.array([1, 0, 0])
+    >>> X_test = np.array([[1.5, 2.3, 5.7, 0, 1]])
+    
+    >>> clf = GeneralNB([
+    >>>     ("gaussian", GaussianNB(), [0, 1, 2]),
+    >>>     ("bernoulli", BernoulliNB(), [3, 4])
+    >>> ])
+    >>> clf.fit(X, y)
+    >>> print(clf.predict(X_test))
+    [1]
+    >>> print(clf.score([[2.7, 3.8, 1, 0, 1]],[0]))
+    1.0
+
+    >>> df = pd.DataFrame(X)
+    >>> df.columns = list("abcde")
+    >>> df["y"] = [1, 0, 0]
+    >>> clf = GeneralNB([
+    >>>     ("gaussian", GaussianNB(), ["a", "b", "c"]),
+    >>>     ("bernoulli", BernoulliNB(), ["d", "e"])
+    >>> ])
+    >>> clf.fit(df.iloc[:,:-1], df["y"])
+
+    >>> df_test = pd.DataFrame(X_test)
+    >>> df_test.columns = list("abcde")
+    >>> print(clf.predict(df_test))
+    [1]
+    >>> print(clf.score([[2.7, 3.8, 1, 0, 1]],[0]))
+    1.0
+    """
+    # TODO consider jll for each estimator
+    # TODO unify variable names with similar meaning 
+    #      ("distribution", "model", "nb"), ("column", "feature")
+
+    def __init__(self, distributions):
+        self.distributions = distributions
+        self.classes_ = None
+        self.n_features_ = None
+        self._fits = None
+        self._columns = None
+        self._df_columns = None
+        self._is_fitted = False
+
+    def fit(self, X, y):
+        """Fit Gaussian Naive Bayes according to X, y
+
+        Parameters
+        ----------
+        X : array-like, shape (n_samples, n_features)
+            Training vectors, where n_samples is the number of samples
+            and n_features is the number of features.
+        y : array-like, shape (n_samples,)
+            Target values.
+
+        Returns
+        -------
+        self : object
+        """
+        self._validate_distributions(X)
+        self._check_X_y(X, y)
+
+        self.classes_ = np.unique(y)
+
+        self._fits = [
+            (nb.fit(_safe_indexing(X, features, axis=1), y), features)
+            for (_, nb, _), features
+            in zip(self.distributions, self._columns)]
+
+        self._is_fitted = True
+
+        return self
+
+    def _joint_log_likelihood(self, X):
+        """Calculate the posterior log probability of the samples X
+
+        Parameters
+        ----------
+        X : ndarray
+        
+        Returns
+        -------
+        jll : ndarray, shape (1, n_classes)
+
+        Raises
+        ------
+        NotFittedError
+            If estimators have not been fitted
+        """
+        if not self._is_fitted:
+            raise NotFittedError("Call the fit() method first "
+                                 "before calling predict().")
+
+        # Obtain the log priors from each fitted estimator
+        log_priors = [
+            nb.class_log_prior_
+            if hasattr(nb, 'class_log_prior_') else np.log(nb.class_prior_)
+            for nb, _ in self._fits]
+
+        # Take any class log prior from the estimators
+        if np.allclose(*log_priors):  
```

Comment:
`allclose` only works with two arguments. So you could instead do something like:
```
all_log_priors = np.hstack(log_priors)
if np.max(np.ptp(all_log_priors, axis=1)) < eps:
    ....
```